2020-02-Spring/0_arrays_matrix/README.md

```diff
@@ -26,21 +26,56 @@ In the style of:
 
 <a name="p1"/>
 
-### 1. PROBLEM 1 TODO :bug:
+### 1. Best Time to Buy and Sell Stock
 
-Source: TODO :bug:
+Source: [LeetCode](https://leetcode.com/problems/best-time-to-buy-and-sell-stock/)
 
 #### Scenario
 
-Problem Statement TODO :bug:
+Say you have an array of n elements representing stock prices.
+
+Price of the stock on the first day is the first element, second day is the second element and so on. 
+
+If you were only permitted to complete at most one transaction (i.e., buy one and sell one share of the stock), design an algorithm to find the maximum profit.
+
+Note that you cannot sell a stock before you buy one.
+
+Note that you can only sell forwards, not backwards. 
 
 #### Example Input
 
-If the problem is simple enough, remove this section. TODO :bug:
+**Input:** \[7,1,5,3,6,4]
+
+**Output:** 5
```

Comment:
So outputting the purchase date doesn't matter? Why? If it doesn't, make that clear in the problem statement.

Reply:
I had to modify the question a bit because the original was kinda confusing. Should I make it more obvious or should I just leave it vague like that so people can kinda practice asking clarifying questions?

Reply:
I personally think this is fine as-is -- currently, the problem statement says "find the maximum profit", but the reader must clarify (or infer) that it is the maximum price difference for a single unit of stock. This isn't inconsistent and is an acceptable amount of vague imo.

For any array-type problem, an interviewer can create problem variations on the fly by changing the return value requirements. (ie "What if you had to return the dates instead of the maximum profit?"). It's up to y'all how you want to incorporate that idea into your sessions.

Reply:
So the reason this is confusing, is because both the sell date and the max profit is 5. So I thought we were returning the date. I think it would be nice if we could change the example a bit, but it would also be OK to say this is a trick you're meant to catch/question.